src/adapter/src/catalog.rs

```diff
@@ -2849,6 +2850,114 @@ impl Catalog {
                         )?;
                     }
                 }
+                Op::RenameSchema {
+                    database_spec,
+                    schema_spec,
+                    new_name,
+                } => {
+                    let ResolvedDatabaseSpecifier::Id(database_id) = &database_spec else {
+                        let err = anyhow::anyhow!("Cannot rename schemas in the ambient database");
+                        return Err(AdapterError::Unstructured(err));
+                    };
+                    let SchemaSpecifier::Id(schema_id) = &schema_spec else {
+                        let err = anyhow::anyhow!("Cannot rename temporary schemas");
+                        return Err(AdapterError::Unstructured(err));
+                    };
```

Comment:
Why is this an error? Either way I'd prefer to use a structured error here.

Reply:
Good thought, I refactored this away